packages/firebase_storage/lib/firebase_storage.dart

```diff
@@ -60,24 +60,134 @@ class StorageReference {
+class StorageMetadata {
+  StorageMetadata({
+    this.cacheControl,
+    this.contentDisposition,
+    this.contentEncoding,
+    this.contentLanguage,
+    this.contentType,
+  }) : bucket = null,
+       generation = null,
+       metadataGeneration = null,
+       path = null,
+       name = null,
+       sizeBytes = null,
+       creationTimeMillis = null,
+       updatedTimeMillis = null,
+       md5Hash = null;
+
+  StorageMetadata._fromMap(Map<String, dynamic> map)
+      : bucket = map['bucket'],
+        generation = map['generation'],
+        metadataGeneration = map['metadataGeneration'],
+        path = map['path'],
+        name = map['name'],
+        sizeBytes=  map['sizeBytes'],
+        creationTimeMillis = map['creationTimeMillis'],
+        updatedTimeMillis = map['updatedTimeMillis'],
+        md5Hash = map['md5Hash'],
+        cacheControl = map['cacheControl'],
+        contentDisposition = map['contentDisposition'],
+        contentLanguage = map['contentLanguage'],
+        contentType = map['contentType'],
+        contentEncoding = map['contentEncoding'];
+
+  /// The owning Google Cloud Storage bucket for the [StorageReference].
+  final String bucket;
+
+  /// A version String indicating what version of the [StorageReference].
+  final String generation;
+
+  /// A version String indicating the version of this [StorageMetadata].
+  final String metadataGeneration;
+
+  /// The path of the [StorageReference] object.
+  final String path;
+
+  /// A simple name of the [StorageReference] object.
+  final String name;
+
+  /// The stored Size in bytes of the [StorageReference] object.
+  final int sizeBytes;
+
+  /// The time the [StorageReference] was created.
+  final num creationTimeMillis;
+
+  /// The time the [StorageReference] was last updated.
+  final num updatedTimeMillis;
+
+  /// The MD5Hash of the [StorageReference] object.
+  final String md5Hash;
+
+  /// The Cache Control setting of the [StorageReference].
+  String cacheControl;
+
+  /// The content disposition of the [StorageReference].
+  String contentDisposition;
+
+  /// The content encoding for the [StorageReference].
+  String contentEncoding;
+
+  /// The content language for the StorageReference, specified as a 2-letter
+  /// lowercase language code defined by ISO 639-1.
+  String contentLanguage;
+
+  /// The content type of the [StorageReference].
+  String contentType;
+
+  Map<String, dynamic> toMap() {
+    return <String, dynamic> {
+      'bucket': bucket,
+      'generation': generation,
+      'metadataGeneration': metadataGeneration,
+      'path': path,
+      'name' : name,
+      'sizeBytes': sizeBytes,
+      'creationTimeMillis': creationTimeMillis,
+      'updatedTimeMillis': updatedTimeMillis,
+      'md5Hash': md5Hash,
+      'cacheControl': cacheControl,
+      'contentDisposition': contentDisposition,
+      'contentLanguage': contentLanguage,
+      'contentType': contentType,
+      'contentEncoding': contentEncoding,
```

Comment:
We ignore most of this information on the platform side. I find it a bit misleading to package up all of it here. Is `toMap` used by anyone except `StorageUploadTask`? If not, then perhaps it can be renamed `toUploadMap` or the responsibility for populating a `Map` with upload metadata can moved to `StorageUploadTask`.